bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+                if self.record.due
+                else None
+            )
+
+        # use deadline as "due".
+        # deadline if set is only a date with no time or timezone. adjust to locla time
+        due = (
+            datetime.combine(self.record.deadline.date, default_time, tzinfo=None)
+            if self.record.deadline
+            else None
+        )
+
+        task = {
+            "project": self.extra["project"],
+            "priority": self.PRIORITY_MAP[self.record.priority],
+            # "annotations": None,
+            "tags": self.record.labels if self.record.labels else [],
+            "scheduled": scheduled,
+            "due": due,
+            "status": "completed" if self.record.is_completed else "pending",
+            "entry": self.record.created_at,
+            self.ID: self.record.id,
+            self.CONTENT: self._unescape_content(self.record.content),
+            self.DESCRIPTION: self._unescape_content(self.record.description),
+            self.DURATION: self.extra["duration"],
+            self.ASSIGNEE: self.extra["assignee"],
+            self.ASSIGNER: self.extra["assigner"],
+            self.SECTION: self.extra["section"],
+            self.URL: self.record.url,
+        }
+        return task
+
+    def get_default_description(self):
+        description = self.build_default_description(
+            title=self._unescape_content(self.record.content),
+            url=self.record.url if self.config.inline_links else '',
+            number=self.record.id,
+            cls="issue",
+        )
+        return description
+
+
+class TodoistService(Service):
+    ISSUE_CLASS = TodoistIssue
+    CONFIG_SCHEMA = TodoistConfig
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.client = TodoistClient(
+            token=self.config.token,
+            filter=self.config.filter,
+        )
+
+    def get_owner(self, issue):
+        # Issue assignment hasn't been implemented yet.
+        raise NotImplementedError(
+            "This service has not implemented support for 'only_if_assigned'."
+        )
+
+    def issues(self):
+        project_index = {project.id: project.name for project in self.client.get_projects()}
+        section_index = {section.id: section.name for section in self.client.get_sections()}
+        user_index = {
+            user.id: f"{user.name} <{user.email}>"
+            for project in project_index.keys()
+            for user in self.client.get_users(project)
+        }
+        for issue_iter in self.client.get_issues():
+            for issue in issue_iter:
+                extra = {
+                    "project": project_index[issue.project_id],
+                    "section": (
+                        section_index[issue.section_id]
+                        if issue.section_id in section_index.keys()
+                        else None
+                    ),
```

Comment:
This will have the same behavior. But is it actually possible for an issue to have a section that doesn't correspond to an active section?
```suggestion
                    "section": section_index.get(issue.section_id),
```

Reply:
much cleaner, making the same change for project, assigner and assignee to use get()